BacktraceScreen.c

```diff
@@ -0,0 +1,301 @@
+#include "BacktraceScreen.h"
+
+#include <execinfo.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/types.h>
+#include <ctype.h>
+#include <unistd.h>
+
+#include "FunctionBar.h"
+#include "InfoScreen.h"
+#include "Panel.h"
+#include "Object.h"
+#include "Vector.h"
+#include "XUtils.h"
+#include "sys/wait.h"
+
+#define DWARF_INFORMATION_NOT_FOUND 1
+
+static const char* const BackTraceScreenFunctions[] = {"Refresh", "Done", NULL};
+static const char* const BackTraceScreenKeys[] = {"F3", "Esc"};
+static const int  BackTraceScreenEvents[] = {KEY_F(3), 27};
+
+static char *BacktraceScreen_parseEuStackFunction(char *current_line) {
+   while (current_line  && isspace(*current_line)) {current_line++;}
+   char *beginOfName = current_line;
+
+   char *backupOfLastSpace = current_line;
+   while (current_line && *current_line != '-') {
+      if (isspace(*current_line))
+         backupOfLastSpace = current_line;
+      current_line++;
+   }
+
+   size_t size_name = (backupOfLastSpace - beginOfName) + 1;
+   char *name = xStrndup(beginOfName, size_name);
+   name[size_name - 1] = '\0';
+   return name;
+}
+
+static char *BacktraceScreen_parseEuStackPath(char *current_line) {
+   while (current_line  && *current_line != '-') {current_line++;}
+   current_line++;
+
+   while (current_line  && isspace(*current_line)) {current_line++;}
+   char *beginOfPath = current_line;
+   while (current_line  && !isspace(*current_line)) {current_line++;}
+
+   size_t sizePath = (current_line - beginOfPath);
```

Comment:
Technically pointer subtraction would result in a type called `ptrdiff_t`, which is signed. If you are casting it to the unsigned type `size_t`, better make it explicit.

Alternatively, you can write the loop in the way that avoids pointer subtraction:
```c
   for (size_t len = 0; true; len++) {
      if (beginOfPath[len] == '\0' || isspace(beginOfPath[len]))
         break;
   }
```

Reply:
Or even more concise …
```c
   for (size_t len = 0; beginOfPath[len]; len++)
      if (isspace(beginOfPath[len]))
         break;
```

@Explorer09 Although that variant would optimize to NOP due to lack of side effects (as does your example); but you get the gist …